client/src/app/tabs/getEditMenu.js

```diff
@@ -302,8 +305,20 @@ export function getDefaultUndoRedoEntries() {
   }];
 }
 
+
 // helpers //////////
 
 function isDefined(value) {
   return value !== undefined;
 }
+
+// @pinussilvestrus: using this helper function instead of chaining
+// the globals.isMac property through the whole application to avoid
+// larger dependencies
+function isMac() {
+  const platform = window.navigator.platform;
+
+  return ['Mac68K', 'MacPPC', 'MacIntel'].indexOf(platform) > -1;
+}

```

Comment:
How about if we could just inject the value at the build time to avoid having to adjust the function in the future? I am wondering what value `navigator.platform` will take on new ARM-based MacBooks.

Reply:
> inject the value at the build time

What would you think about in particular?

Reply:
> How about if we could just inject the value at the build time to avoid having to adjust the function in the future? I am wondering what value navigator.platform will take on new ARM-based MacBooks.

This is a whole new issue that we should investigate independently. I've created an issue for it: https://github.com/camunda/camunda-modeler/issues/1996

I'm looking into https://github.com/camunda/camunda-modeler/blob/develop/client/src/app/AppParent.js#L106 and am wondering if we can have the whole hack thing (mac / is not mac) in one place.

I could imagine that the keyboard shortcuts returns _updated_ menu shortcuts and applies the relevant changes (i.e. swaps `Backspace` with `Delete` if isMac is there. This way we can keep the actual menu definition (`getEditMenu` short and simple).

Reply:
I like this idea a lot! Lemme have a look.